internal/pkg/workspace/workspace.go

```diff
@@ -60,16 +62,20 @@ const (
 // Summary is a description of what's associated with this workspace.
 type Summary struct {
 	Application string `yaml:"application"` // Name of the application.
-
-	Path string // absolute path to the summary file.
+	Path        string `yaml:"-"`           // absolute path to the summary file.
 }
 
 // Workspace typically represents a Git repository where the user has its infrastructure-as-code files as well as source files.
 type Workspace struct {
 	workingDirAbs string
 	copilotDirAbs string
-	fs            *afero.Afero
-	logger        func(format string, args ...interface{})
+
+	summary       *Summary
+	summaryErr    error
```

Comment:
I'm probably missing some thing important - why do we want to store `summaryErr` in `workspace` ?

Reply:
We're now using [`sync.Once`](https://pkg.go.dev/sync#Once) in the [Summary method](https://github.com/aws/copilot-cli/pull/4202/files/c47847b0d5e92c2b67851f261c478462bb28aa2a#diff-8646fd3cdd4c58b0a1c3c3c09348d53c9fe5d6231b922f87f91c02f90b57f8d2R161) to read the `.workspace` file only once.
So we need to store its results in the struct: the successful parsed file or any error that occurred while reading the file.
This way, when `Summary` is invoked again then we can return the [saved values](https://github.com/aws/copilot-cli/pull/4202/files/c47847b0d5e92c2b67851f261c478462bb28aa2a#diff-8646fd3cdd4c58b0a1c3c3c09348d53c9fe5d6231b922f87f91c02f90b57f8d2R177).

If let's say we don't save the error, say we have:
```go
var summaryErr error
summarizeOnce.Do(func() {
    // existing logic...
})
return ws.summary, summaryErr
```
Then the first time a caller invokes `Summary` we will return an error appropriately, but the second time the error will be `nil` which is not what we want.